### ISSUE TITLE
try_fopen destroys error handler in specific contexts

### ISSUE THREAD

**Issue by SunflowerFuchs** [guzzle/psr7#318]
**PHP version**: 7.4.2 

**Description**
(See https://github.com/getsentry/sentry-php/issues/976 for additional info.)
If multiple errorhandlers are set and try_fopen is getting called from within one of them, then the set_error_handler/restore_error_handler combination for catching fopen errors does not work.
https://github.com/guzzle/psr7/blob/4ae8647dd962ac404d578fd42e00ffc81cb00d75/src/functions.php#L311-L321

**How to reproduce**
I've set up a reproducible example for sentry at https://github.com/SunflowerFuchs/sentry-error, but if you want, I can write a more minimal example for this later.

**Possible Solution**
Replacing the code with a simple try/catch would work, but that would mean losing the `func_get_args()[1]` from within the error handler.

**Comment by GrahamCampbell** [comment]
> Replacing the code with a simple try/catch would work

No, it would not. Errors that are "raised" are not the same as "exceptions".

**Comment by GrahamCampbell** [comment]
It is my understanding that PHP fixed the error handler stacking issue a few months ago anyway, so as long as you're using a supported version of PHP, or an enterprise version with this backported, you should no longer be running into this issue.

**Comment by GrahamCampbell** [comment]
Yep, here we go: https://github.com/php/php-src/pull/5206. Fixed in PHP 7.3.16, 7.4.4.
